File: apps/mobile/test/languages_test.dart
import 'package:flutter/widgets.dart';
import 'package:flutter_test/flutter_test.dart';
import 'package:voxeli/shared/languages.dart';

void main() {
  test('registry has unique codes and marks RTL scripts', () {
    final codes = kLanguages.map((l) => l.code).toList();
    expect(codes.toSet().length, codes.length);
    for (final c in ['he', 'ar', 'fa', 'ur', 'yi']) {
      expect(directionOf(c), TextDirection.rtl, reason: c);
    }
    expect(directionOf('en'), TextDirection.ltr);
    expect(directionOf('unknown'), TextDirection.ltr);
    expect(languageByCode('he-IL')?.code, 'he');
  });
}

File: packages/domain/src/languages.test.ts
import { describe, expect, it } from 'vitest';
import { LANGUAGES, directionOf, getLanguage, isRtl, isUiLocale } from './languages.js';

describe('language registry', () => {
  it('has unique codes', () => {
    const codes = LANGUAGES.map((l) => l.code);
    expect(new Set(codes).size).toBe(codes.length);
  });
  it('marks Hebrew, Arabic, Persian, Urdu, Yiddish as RTL', () => {
    for (const c of ['he', 'ar', 'fa', 'ur', 'yi']) expect(isRtl(c)).toBe(true);
    for (const c of ['en', 'de', 'ru', 'ja']) expect(isRtl(c)).toBe(false);
  });
  it('resolves regional variants to primary subtag', () => {
    expect(getLanguage('he-IL')?.code).toBe('he');
    expect(getLanguage('en-US')?.code).toBe('en');
    expect(getLanguage('zh-Hant')?.code).toBe('zh-Hant');
  });
  it('defaults unknown languages to ltr without throwing', () => {
    expect(directionOf('xx')).toBe('ltr');
  });
  it('exposes the seven launch UI locales', () => {
    expect(isUiLocale('he')).toBe(true);
    expect(isUiLocale('ja')).toBe(false);
  });
});
